omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DiagnosisResource2_2.java

```diff
@@ -0,0 +1,93 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_2;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DataDelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+
+/**
+ * {@link Resource} for Diagnosis, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + "/patientdiagnoses", supportedClass = Diagnosis.class, supportedOpenmrsVersions = "2.2.*")
+
+public class DiagnosisResource2_2 extends DataDelegatingCrudResource<Diagnosis> {
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Diagnosis getByUniqueId(String s) {
+		return Context.getDiagnosisService().getDiagnosisByUuid(s);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#delete(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected void delete(Diagnosis diagnosis, String reason, RequestContext requestContext) throws ResponseException {
+		Context.getDiagnosistService().voidDiagnosis(diagnosis, reason);
+	}
+
+	@Override
+	public Diagnosis newDelegate() {
+		return new Diagnosis();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#save(java.lang.Object)
+	 */
+	@Override
+	public Diagnosis save(Diagnosis diagnosis) {
+		return Context.getDiagnosisService().saveDiagnosis(diagnosis);
+	}
+
+	@Override
+	public void purge(Diagnosis diagnosis, RequestContext requestContext) throws ResponseException {
+		Context.getDiagnosisService().purgeService(diagnosis);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getRepresentationDescription(org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+		if (representation instanceof DefaultRepresentation) {
```

Comment:
Can you also add for the full representation?

Reply:
yeah sure